File: ChatGPT/Direct_Attempt/python/output/output_13a.py
# ChatGPT-generated solution will go here.
def parse_input():
    machines = []
    with open("input.txt") as f:
        lines = [line.strip() for line in f if line.strip()]
        for i in range(0, len(lines), 3):
            ax, ay = map(int, lines[i].split(":")[1].split(","))
            bx, by = map(int, lines[i+1].split(":")[1].split(","))
            px, py = map(int, lines[i+2].split(":")[1].split(","))
            machines.append(((ax, ay), (bx, by), (px, py)))
    return machines

def extract_values(line):
    parts = line.split(",")
    x = int(parts[0].split("+")[1] if '+' in parts[0] else parts[0].split("=")[1])
    y = int(parts[1].split("+")[1] if '+' in parts[1] else parts[1].split("=")[1])
    return x, y

def main():
    machines = []
    with open("input.txt") as f:
        lines = [line.strip() for line in f if line.strip()]
        for i in range(0, len(lines), 3):
            ax, ay = extract_values(lines[i])
            bx, by = extract_values(lines[i+1])
            px, py = extract_values(lines[i+2])
            machines.append(((ax, ay), (bx, by), (px, py)))

    max_presses = 100
    total_cost = 0
    for (ax, ay), (bx, by), (px, py) in machines:
        min_cost = float('inf')
        for a in range(max_presses + 1):
            for b in range(max_presses + 1):
                x = a * ax + b * bx
                y = a * ay + b * by
                if x == px and y == py:
                    cost = a * 3 + b * 1
                    if cost < min_cost:
                        min_cost = cost
        if min_cost != float('inf'):
            total_cost += min_cost

    print(total_cost)

if __name__ == "__main__":
    main()

File: ChatGPT/clojure/output/output_13a.clj
;; ChatGPT-generated solution will go here.
(ns output_13a
  (:require [clojure.string :as str]))

(defn parse-machine [chunk]
  (let [nums (map #(Long/parseLong (re-find #"-?\d+" %)) chunk)
        [[ax ay] [bx by] [px py]] (partition 2 nums)]
    {:a [ax ay], :b [bx by], :p [px py]}))

(defn parse-input [lines]
  (->> lines
       (partition-by str/blank?)
       (remove #(= (first %) ""))
       (map parse-machine)))

(defn try-machine [{:keys [a b p]}]
  (let [[ax ay] a
        [bx by] b
        [px py] p
        max-presses 100]
    (loop [x 0, best nil]
      (if (> x max-presses)
        best
        (let [rx (- px (* x ax))
              ry (- py (* x ay))]
          (if (and (zero? bx) (not (zero? rx)))
            (recur (inc x) best)
            (if (and (zero? by) (not (zero? ry)))
              (recur (inc x) best)
              (let [b-x (if (zero? bx) nil (/ rx bx))
                    b-y (if (zero? by) nil (/ ry by))]
                (if (and b-x b-y (= b-x b-y) (<= 0 b-x max-presses) (== (long b-x) b-x))
                  (let [y (long b-x)
                        cost (+ (* x 3) y)]
                    (recur (inc x) (if (or (nil? best) (< cost best)) cost best)))
                  (recur (inc x) best)))))))))

(defn part [lines]
  (let [machines (parse-input lines)
        token-costs (keep try-machine machines)]
    (reduce + token-costs)))

(defn -main [& args]
  (let [filename (first args)
        lines (str/split-lines (slurp filename))]
    (println (part lines))))
